File: Code_Forces/line_trip.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

int main(){
    // freopen("input.txt", "r", stdin);
    int t;
    cin>>t;
    while(t--){
        //Take Inputs
        int n,x;
        cin>>n>>x;

        //Write logic for finding maxi

        

        vector<int> ans(n); //Impletation with array to be done and python also
        
        for(int i=0;i<n;i++){
            cin>>ans[i];
        }
        int maxi = ans[0];
        int last_maxi = 2 * (x - ans[n-1]);

        for(int i=1;i<n;i++){
            maxi = max(ans[i] - ans[i-1],maxi);
        }

        //Print maxi
        cout<<max(maxi,last_maxi)<<endl;
        
    }
    return 0;
}

File: Code_Forces/misc/line_trip.py
t = int(input())

while(t):
    t = t-1
    n,x = map(int,input().split())

    arr = [] * n

    for i in range(0,n):
        arr.append(int(input()))
    
    maxi = arr[0]
    last_maxi = 2 * (x - arr[n-1])

    for i in range(1,n):
        maxi = max(maxi,arr[i] - arr[i-1])
    
    print(max(maxi,last_maxi))


